### PR TITLE
Add a driver for the ADIS16448 IMU

### PR DESCRIPTION
**Pull request by ASM3**
This driver allows interfacing the AD's ADIS16448 IMU via the external SPI port and the usage of its gyroscope, accelerometer and magnetometer by using an adapter interface.

### CODE REVIEW

**Comment by dagar** [comment]
Overall looks good. What are the additional IOCTLs for?

**Comment by dagar** [comment]
CI failed due to style errors (run make format) and this pushed the px4fmu-v2_default over the flash limit. You could add it to px4fmu-v3_default and commented out in px4fmu-v2_default.

**Comment by ASM3** [comment]
for internal IMU settings (dynamic range, internal filter, sampling rate, etc)

**Comment by ASM3** [comment]
Additional IOCTLs where added for obtaining the device product ID, which is described with a unique product ID number of 0x4040 (16448) for configuration purpose.

**Comment by ASM3** [comment]
How come continuous-integration fails because of ADIS16448 driver on nuttx_px4fmu-v2_default? (ADIS16448 driver is commented out on px4fmu-v2_default)

File: src/drivers/adis16448/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+############################################################################
+px4_add_module(
+	MODULE drivers__adis16448
+	MAIN adis16448
+	STACK_MAIN 1200
+	COMPILE_FLAGS
+		-Os
```
Comment (dagar) [inline src/drivers/adis16448/CMakeLists.txt]: You can remove the optimization setting.

**Comment by dagar** [comment]
Would you mind rebasing? There's been some churn in the driver layer to handle compatibility issues between all the pixhawk variants that were supposed to be binary compatible (but weren't).

src/drivers/device/spi.h moved to src/drivers/device/nuttx/SPI.hpp

Once it's rebased and CI is happy we can merge this immediately.

**Comment by dagar** [comment]
Thanks!

**Comment by jinchengde** [comment]
that‘s looks great!

**Comment by ASM3** [comment]
Yes I am currently working on this

**Comment by huotan** [comment]
Impressive!
Look forward to your follow-up work。
@ASM3

**Pull request merged**